lib/MadNLPGPU/src/cudss.jl

```diff
@@ -0,0 +1,104 @@
+import CUDSS
+import SparseArrays
+
+@kwdef mutable struct CudssSolverOptions <: MadNLP.AbstractOptions
+    cudss_algorithm::MadNLP.LinearFactorization = MadNLP.BUNCHKAUFMAN
```

Comment:
I don't like the name `BUNCHKAUFMAN`. I'm talking about it with Mike and we don't even if it exists for sparse decomposition. LDL and LBL are more relevant names I think (and shorter!).

Update: From my ChatGPT available in Hong Kong (Mike), the authors of the paper "Accurate Symmetric Indefinite Linear Equation Solvers" (SIAM) conclude that we can't do Bunch-Kaufman for sparse matrices because it's not stable.

Reply:
Thanks, Mike for the comments! Indeed LDL was more proper here